### ISSUE TITLE
target.tweakXY function (MEC)

### ISSUE THREAD

**Issue by silkenelson** [pcdshub/Bug-Reports-and-Requests#16]
**Feature Request**
this function is missed. I suspect you tweak two motors using left/right and up/down arrows. definitely need tweaking (with having left/right arrow change value and up/down change step size. Generalize to a 2-d tweak w/ fixed tweak steps.

**Existing Alternatives**
Use old python. Using GUI is a lot more hassle.

**Context**
Code exists in old python.

**Comment by ZLLentz** [comment]
2d tweak already exists. Which devices need this added onto them?

**Comment by silkenelson** [comment]
@slactjohnson can you check if target.tweakXY exists? I don't know exactly what MEC means with that.

**Comment by ZLLentz** [comment]
The function definitely exists, but I would 100% believe that there are devices that want to use it that don't have it set up.

**Comment by slactjohnson** [comment]
@silkenelson It is basically what you described. I believe they had object properties that contained the tweak values, and then up/down left/right arrows move the stages by the tweak values.

**Comment by egaltier** [comment]
Indeed, we were using the keyboard arrows as joystick to fine tune the target alignment to the interaction point. In addition, we were using the key '+' to double the tweak value and '-' to divide by 2  so that we could do larger motion quickly if needed. 'q' was used to quit the tweaking display since the RBV values were displayed in the python session when the movement was being done.

**Comment by ZLLentz** [comment]
@slactjohnson there is a function here that does most of the lifting so we don't have to rewrite the whole thing:
https://github.com/pcdshub/pcdsdevices/blob/b43da719a36231859abd167a19addee04a8ec5a8/pcdsdevices/interface.py#L1054

The 2d keystrokes are a bit different than in the old python. We should either change them or make them configurable somehow.

**Comment by slactjohnson** [comment]
@ZLLentz Thanks! My first inclination would be that we leave the existing implementation you linked to as is, and show MEC how to use it. I don't see why the keystrokes need to be different in MEC vs other hutches.

**Comment by ZLLentz** [comment]
I agree that it's good for them all to be the same. I'm not sure that the strokes I picked are the ones that XPP/XCS/etc. were expecting.

**Comment by silkenelson** [comment]
I am wondering that as well. I could ask the scientists and we can decide whose keystrokes win ;) 
They should be the same for all hutches just to avoid mistakes being made. The difference is how the tweak values get adjusted, so at least not the main operation.

**Comment by ZLLentz** [comment]
My opinion is that the keystrokes Eric cited above are better than the ones implemented now since there is lower likelihood of accidentally moving the motor when you meant to change the step size.

E.g. I prefer +/- to shift+up, shift+down
But of course I am not the end user so I defer judgement

**Comment by slactjohnson** [comment]
I'm still not clear on how you increase/decrease the step on a particular axis with +/-. Or did you simply increase/decrease the step size of both axes at the same time? @egaltier? 

I kind of like the shift implementation since it seems more flexible (i.e. you can clearly address each axis separately), and the mapping feels intuitive to me. But I see the issue with accidentally moving the motor.

**Comment by egaltier** [comment]
When using the 'old' `target.tweakXY()`, pushing '+' or '-' was changing the step by a factor of 2 for both axis. It is quick enough to hit the '+' and '-'  that it is not a problem to only have one shortcut for both axis. At least this is the feeling we have when we used it ^^;

**Comment by silkenelson** [comment]
FWIW, the surveyed HXR/XCS scientists are willing to adjust what they use for the 2-d tweak. A slight preference for +/- was mentioned due to the reduced likelihood of hitting the wrong key.
